BrowserKit/Sources/Redux/Store.swift

```diff
@@ -0,0 +1,73 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// Stores your entire app state in the form of a single data structure.
+/// This state can only be modified by dispatching Actions to the store.
+/// Whenever the state of the store changes, the store will notify all store subscriber.
+public class Store<State: StateType>: DefaultDispatchStore {
```

Comment:
Similar here, does the store need to be a class or can it be a struct?

Reply:
Here the state and the subscription change so some function will need to be annotated with `mutating`

Reply:
This has me a little worried because this needs to be very thread safe. Lets hop on a call tomorrow and discuss.